Sources/KlaviyoSwift/KlaviyoState.swift

```diff
@@ -487,3 +503,14 @@ extension Profile {
         return profile
     }
 }
+
+extension String {
+    fileprivate func isNotEmptyOrSame(as state: String?, identifier: String) -> Bool {
```

Comment:
Since I was doing this thrice for three of the identifier setters and then again thrice when setting a profile, I instead created an fileprivate extension on string and calling this to keep the comparison logic in just one location. Call site now looks like - `email.isNotEmptyOrSame(as: self.email, identifier: "email")`

Reply:
yeah this is nice.